cpp/src/Slice/SliceUtil.cpp

```diff
@@ -450,7 +450,7 @@ Slice::argvToArgs(int argc, char* argv[])
      {
          ids.push_back(scoped.substr(next));
      }
-     else
+     else if(allowEmpty)
      {
          ids.push_back("");
```

Comment:
I don't remember why we need to add this empty id!

Reply:
Me either!